_posts/2018-02-28-smart-window-shades.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: Smart Window Shades
+date: 2018-03-01 00:00:00
+categories: projects
+description: A web service that allows one to control their window shades using a website
+author: Ayush Goel
+author-image: https://avatars2.githubusercontent.com/u/10407155?s=460&v=4
+author-bio: First Year Computer Science Student at Rochester Institute of Technology
+author-email: axg4975@rit.edu
+author-social:
+  github: https://github.com/axg4975
+  linkedin: https://www.linkedin.com/in/ayush-goel-589ba215a/
+---
+# Summary
+Smart Shades is a web service that allows one to control window shades using a website. The hardware requirement is a Raspberry Pi (connected to the Internet), a Stepper Motor, housing for the stepper motor (3D printed), a Stepper Motor Driver (I used the Pololu A4988), a 10K Ohm Resistor, and a domain. The application uses python, Flask, OpenShift in the back end, Bootstrap, JavaScript, and Ajax. In short, there are two Flask apps running, one on [CSH domain](shades.csh.rit.edu) and one on the Pi. The flask app on CSH website has the JavaScript and Ajax to make requests to the Flask app on the Pi, which can then take care of changing the window shades.
+
+# The Story
+I picked up this IOT project from R&D, Spencer had the hardware and asked if someone could do the software by ImagineRIT. I had some ideas for ImagineRIT, but this one seemed more fun than the rest of them. I went into this knowing how to code, specifically in python, but not having a great understanding of how to incorporate the hardware with the software, or how the web stuff works. I started the project by figuring out how to use RPi.GPIO library by making a simple program with LEDs. Then I started using the Stepper Motor Driver and the Stepper Motor, which this ended up being a hardware problem not a software one. Fun fact, not having documentation makes everything much more difficult. After countless hours I figured out that I can connect an LED to 2 of the 4 pins of the motor and if the led lights up upon turning the motor then I have found a pair of wires. From there I had to figure out how stepper motors work in software, which comparatively was not that difficult.
```

Comment:
* First sentence is a comma splice.
* `s/how to code, specifically in python/use Python,`
* "the web stuff" is too conversational, try "web application programming"
* `Then I started using the Stepper Motor Driver and the Stepper Motor, which this ended up being a hardware problem not a software one.` -> `Then I started using the stepper motor driver and the stepper motor. This ended up being a hardware problem, not a software one.`
* Strike `Fun fact, not having documentation makes everything much more difficult.`
* Consistently capitalize LED